src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -186,12 +186,21 @@ def _preprocess_data(self, data: List, attr: str) -> List[Union[int, float]]:
 
     def _calculate_mean(self, data: List[Union[int, float]], attr: str) -> None:
         avg = np.mean(data)
+        if self._is_time_field(attr):
+            avg = self._scale(float(avg))
         self._stats_dict[attr]["avg"] = float(avg)
         setattr(self, "avg_" + attr, avg)
 
     def _calculate_percentiles(self, data: List[Union[int, float]], attr: str) -> None:
         p25, p50, p75 = np.percentile(data, [25, 50, 75])
         p90, p95, p99 = np.percentile(data, [90, 95, 99])
+        if self._is_time_field(attr):
+            p25 = self._scale(float(p25))
```

Comment:
It feels like the single location should just be during the output though

Reply:
My understanding was that any export of the data going forward was to be updated to ms from ns.
Today we export in 3 forms: console output, csv, and json. That could grow in the future. 
Your suggestion would  require scaling 3 times instead of at the data source.

If you still think this is the wrong approach, i will change it. I am just not convinced currently, that scaling at the output location is the right call.

Reply:
Refactored to a single location. Thanks for the suggestion @nv-hwoo